src/workerd/jsg/jsg.h

```diff
@@ -362,6 +362,11 @@ class JsExceptionThrown: public std::exception {
 // Use inside a JSG_RESOURCE_TYPE block to declare a property on this object that should be
 // accessible to JavaScript. `name` is the JavaScript member name, while `getter` and `setter` are
 // the names of C++ methods that get and set this property.
+//
+// DO NOT USE unless you have a very good reason. Use JSG_PROTOTYPE_PROPERTY instead. Instance
+// properties are rarely what you want, and using them breaks some V8 optimizations. For example,
+// any object with an instance proprety will not be possible to collect during minor GCs, only
+// major GCs.
```

Comment:
Non-blocking nit: I'm not sure "DO NOT USE" is quite nuanced enough here without some insight into what is a "very good reason". 

I'm not sure I would say "DO NOT USE" at all here. Instead, just document that use of instance properties breaks v8 optimizations and leave it at that. 

Also, a similar comment should like to be added to `JSG_LAZY_INSTANCE_PROPERTY`.

This might also be something worth discussing the jsg/README.md in a bit more detail.

Reply:
When are instance properties ever preferred, though?

Reply:
I changed the wording, does it look OK now?